lib/llvmIr.js

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2018, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+const utils = require('./utils');
+
+class LlvmIrParser {
+    constructor(compilerProps) {
+        // @todo More settings/compiler properties
+
+        this.maxIrLines = 500;
+        if (compilerProps) {
+            this.maxIrLines = compilerProps('maxLinesOfAsm', this.maxIrLines);
+        }
+    }
+
+    trimIrLine(line, filters) {
+        if (!filters.trim) return line;
+        const splat = line.split(/\s+/);
+        if (splat[0] === "") {
```

Comment:
Isn't this on utils.js too? You'd only need to forward the args when the filters ask for it

Reply:
I copied `lib/asm.js` and modified it, so that's where I got it from.
It doesn't seem to be included in `lib/utils.js`.

Reply:
Sure thing! Thanks for reviewing my code :)
(I renamed `llvmIr.js` to `llvm-ir.js` to stick with conventions.)